app/models/announcement.rb

```diff
@@ -1,32 +1,31 @@
 class Announcement < ActiveRecord::Base
-  attr_accessible :start_date, :end_date, :button, :link, :body
-  
+  attr_accessible :id, :start_date, :end_date, :button, :link, :body
+
   validates :button, presence: true
   validates :link, presence: true
   validates :body, presence: true
-  validates :start_date, presence: true, uniqueness: true
-  validates :end_date, presence: true, uniqueness: true
-  validate :overlapping_dates, on: :create
+  validates :start_date, presence: true
+  validates :end_date, presence: true
+  validate :overlapping_dates
   validate :start_date_before_end_date
 
-  def date_range
-    start_date..end_date
-  end
-
   def overlapping_dates
-    announcements = Announcement.all
-    errors.add(:base, "Announcement conflicts with other announcements") if announcements.any? { |announcement| announcement.date_range.overlaps? date_range }
+    if start_date && end_date
+      Announcement.find do | d |
+        if (start_date <= d.end_date) && (end_date >= d.start_date) && (id != d.id)
+          errors.add(:base, "Conflicts with other Announcements " + d.start_date.to_s + ", " + d.end_date.to_s)
+        end
+      end
+    end
   end
 
   def start_date_before_end_date
-    errors.add(:end_date, "cannot be earlier than start date") if start_date > end_date
+    if start_date && end_date && start_date > end_date
+      errors.add(:end_date, "cannot be earlier than start date") 
+    end
   end
-  
+
   def self.current
```

Comment:
Rails has some built in functionality for this sort of thing called `scopes`: http://guides.rubyonrails.org/active_record_querying.html#scopes